plugin/src/cli_utils/upgrade/mod.rs

```diff
@@ -0,0 +1,258 @@
+                                        ..Default::default()
+                                    }),
+                                    ..Default::default()
+                                }),
+                                ..Default::default()
+                            },
+                            EnvVar {
+                                // Ref: https://github.com/helm/helm/blob/main/cmd/helm/helm.go#L76
+                                name: "HELM_DRIVER".to_string(),
+                                value: env::var("HELM_DRIVER").ok(),
+                                ..Default::default()
+                            },
+                        ]),
+                        liveness_probe: Some(Probe {
+                            exec: Some(ExecAction {
+                                command: Some(vec![
+                                    "pgrep".to_string(),
+                                    "upgrade-job".to_string(),
+                                ]),
+                            }),
+                            initial_delay_seconds: Some(10),
+                            period_seconds: Some(60),
+                            ..Default::default()
+                        }),
+                        volume_mounts: Some(vec![VolumeMount {
+                            read_only: Some(true),
+                            mount_path: "/upgrade-config-map".to_string(),
+                            name: "upgrade-config-map".to_string(),
+                            ..Default::default()
+                        }]),
+                        ..Default::default()
+                    }],
+                    service_account_name: Some(format!("{release_name}-upgrade-service-account-{version}", release_name = release_name.as_str(), version = upgrade_obj_suffix())),
+                    volumes: Some(vec![Volume {
+                        name: "upgrade-config-map".to_string(),
+                        config_map: Some(ConfigMapVolumeSource {
+                            name: Some(format!("{release_name}-upgrade-config-map-{version}", release_name = release_name.as_str(), version = upgrade_obj_suffix())),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    }]),
+                    ..Default::default()
+                }),
+            },
+            ..Default::default()
+        }),
+        ..Default::default()
+    }
+}
+
+pub async fn apply_upgrade(namespace: &str, args: &UpgradeArgs, k8s_client: Client) -> Result<()> {
+    let release_name = match args.release_name.as_ref() {
+        Some(name) => name.clone(),
+        None => helm_release_name(namespace, args.helm_storage_driver.as_str()).await?
+    };
+
+    delete_older_upgrade_events(release_name.as_str(), "upgrade", Api::namespaced(k8s_client.clone(), namespace)).await?;
+
+    create_upgrade_resources(args, release_name.as_str(), namespace, k8s_client).await?;
+
+
```

Comment:
is this formatted? There's a lot of new lines here.

Reply:
It is now..